physical/zookeeper.go

```diff
@@ -52,16 +53,63 @@ func newZookeeperBackend(conf map[string]string) (Backend, error) {
 		machines = "localhost:2181"
 	}
 
+    // zNode owner and schema.
+    var owner string
+    var schema string
+    var schemaAndOwner string
+    schemaAndOwner, ok = conf["znode_owner"]
+    if !ok {
+        owner = "anyone"
+        schema = "world"
+    } else {
+        parsedSchemaAndOwner := strings.SplitN(schemaAndOwner, ":", 2)
+        if !(len(parsedSchemaAndOwner)==2) {
+            return nil, fmt.Errorf("znode_owner expected format is 'schema:owner'")
+        } else {
+            schema = parsedSchemaAndOwner[0]
+            owner = parsedSchemaAndOwner[1]
+        }
+    }
+
+    acl := []zk.ACL{{zk.PermAll, schema, owner}} 
+
+
+    // Authnetication info
+    var schemaAndUser string
+    schemaAndUser, ok = conf["auth_info"]
+    if !ok {
+        owner = ""
+        schema = ""
+    } else {
+        parsedSchemaAndUser := strings.SplitN(schemaAndUser, ":", 2)
+        if !(len(parsedSchemaAndUser)==2) {
+            return nil, fmt.Errorf("auth_info expected format is 'schema:auth'")
+        } else {
+            schema = parsedSchemaAndUser[0]
+            owner = parsedSchemaAndUser[1]
+        }
+    }   
+
+
 	// Attempt to create the ZK client
 	client, _, err := zk.Connect(strings.Split(machines, ","), time.Second)
 	if err != nil {
 		return nil, fmt.Errorf("client setup failed: %v", err)
 	}
 
+    // If auth_info provided - attempt to authenticate 
+    if owner != "" {
```

Comment:
I believe this should check both the owner and schema for non-empty values, since schema is used the line below.